src/main/java/Parser.java

```diff
@@ -64,95 +76,203 @@ public boolean createIndex(String file, String indexDirectoryLocation) throws IO
+
+                            Document luceneDoc = new Document();
+                            boolean empty = true;
+
+                            String docNo = eElement.getElementsByTagName("DOCNO").item(0).getTextContent().replaceAll("\n(?!\n)", " ").trim();
+                            // System.out.println(docNo);
+                            if (!docNo.isEmpty()) {
+                                luceneDoc.add(new TextField(FieldNames.DOCNO.getName(), docNo, Field.Store.YES));
+                            }
+
+                            NodeList headlineField = eElement.getElementsByTagName("HEADLINE");
+                            String headline = "";
+                            if(headlineField.getLength() > 0)
+                                 headline = headlineField.item(0).getTextContent().replaceAll("\n(?!\n)", " ").trim();
+                            // System.out.println(headline);
+                            if (!headline.isEmpty()) {
+                                luceneDoc.add(new TextField(FieldNames.HEADLINE.getName(), headline, Field.Store.YES));
+                                empty = false;
+                            }
+
+                            NodeList textField = eElement.getElementsByTagName("TEXT");
+                            String text = "";
+                            if(textField.getLength() > 0)
+                                text = textField.item(0).getTextContent().replaceAll("\n(?!\n)", " ").trim();
+                            // System.out.println(text);
+                            if (!text.isEmpty()) {
+                                luceneDoc.add(new TextField(FieldNames.TEXT.getName(), text, Field.Store.YES));
+                                empty = false;
+                            }
+
+                            if (!empty) {
+                                documents.add(luceneDoc);
+                            }
+                        }
+                        indexWriter.addDocuments(documents);
+                    }
                 }
             }
-        }
-        catch (InvalidPathException | NoSuchFileException exception){
-            System.out.println("Invalid path! Can not index: " + file);
-            indexWriter.close();
-            directory.close();
+            return true;
+        } catch (Exception e) {
+            // TODO
+            e.printStackTrace();
             return false;
         }
-    
-        //add the created documents to the index and close everything
-        indexWriter.addDocuments(documents);
-        indexWriter.close();
-        directory.close();
-        System.out.println("Indexing finished successfully");
+    }
+
+    /**
+     * Generic function to recursively get all the files to be indexed from a directory
+     * @param dir Files are searched in this directory
+     * @return List of all files to be indexed
+     */
+
+    private static ArrayList<File> getFilesFromDir(File dir){
```

Comment:
I think it's better to make methods static when possible if they are just called from the class rather than an instantiated object. Java tends to optimize memory usage in that case. Also it tends to be good practice to make methods private if they don't need to be called from other classes.

Reply:
ok makes sense, thanks.

Reply:
Even I had the same thought process as @jackgilbride999. @OvrK-tcd What do you think?